### PR TITLE
[SQL] Fix for server create in SQL 0.28

### PR DESCRIPTION
**Pull request by strehan1993**
**Description**

We should not pass tenant Id at all times, since if external admin is missing tenant id will block create. Fixed it in the PR.
Fix for regression - **must go in S189**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by evelyn-ys — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/sql/custom.py [hunk 1]
```diff
@@ -3447,12 +3447,16 @@ def server_create(
     if enable_ad_only_auth:
         ad_only = True
 
+    tenant_id = None
```
Comment (ericshape) [inline src/azure-cli/azure/cli/command_modules/sql/custom.py:3450]: NIT: not pythonic~ LOL
tenant_id=_get_tenant_id() if external_admin_name is not None else None

Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/custom.py:3450]: azdev style sql
azdev linter sql 

pass :)

Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/custom.py:3450]: Style asked me to switch to `is not None` :
image

This is the exact reason I left Data Science track: 
tenant_id=_get_tenant_id() if external_admin_name is not None else None does not even seem like code, I mean it's like writing a story. Not readable at all.

Comment (ericshape) [reply src/azure-cli/azure/cli/command_modules/sql/custom.py:3450]: LOL~ that's python~ LMAO

but yes, if any logic is used twice, it should be abstracted as a method~

**Pull request merged**